### PR TITLE
DAOS-15261 test: update test to use valid user/group

### PR DESCRIPTION
**Pull request by mchaarawi**
Update deployment/io_sys_admin.py to use the correct syntax for `useradd` and `groupadd`.
Also create these on the runner node.

Quick-Functional: true
Test-tag: test_io_sys_admin

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/deployment/io_sys_admin.yaml [hunk 1]
```diff
@@ -56,8 +56,8 @@ container:
   properties: cksum:crc16,cksum_size:16384,srv_cksum:on
   control_method: daos
 container_acl:
-  new_user: daos_ci_tester_1@
-  new_group: daos_ci_test_grp_1@
+  new_user: daos_ci_tester_1
+  new_group: daos_ci_test_grp_1
```
Comment (mchaarawi) [inline src/tests/ftest/deployment/io_sys_admin.yaml:60]: oh so those users do exist, but we are just adding an extra @ causing the issue?

Comment (daltonbohning) [reply src/tests/ftest/deployment/io_sys_admin.yaml:60]: A secondary problem is that when `useradd` fails, the test silently skips over it :)

Comment (mchaarawi) [reply src/tests/ftest/deployment/io_sys_admin.yaml:60]: > But the DAOS ACL entry DOES need the `@`

I thought that was fixed to make it optional..

Comment (daltonbohning) [reply src/tests/ftest/deployment/io_sys_admin.yaml:60]: I didn't realize. I was just modeling after some other tests. I'll make a note / ticket to address that here and other places. Thanks!

**Comment by daltonbohning** [comment]
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13854/5/artifact/Functional%20Hardware%20Medium/deployment/io_sys_admin.py/job.log
Test is working as expected now
```
2024-02-28 02:15:30,017 output           L0696 DEBUG| [stdout]      =Clients/hosts wolf-[149,253], exec cmd: sudo /usr/sbin/useradd -r daos_ci_tester_1
2024-02-28 02:15:30,017 output           L0696 DEBUG| [stdout] 

2024-02-28 02:15:30,233 output           L0696 DEBUG| [stdout]      =Clients/hosts wolf-[149,253], exec cmd: sudo /usr/sbin/groupadd  daos_ci_test_grp_1
2024-02-28 02:15:30,233 output           L0696 DEBUG| [stdout]
...
2024-02-28 02:15:39,814 process          L0604 INFO | Running '/usr/bin/daos container set-owner TestPool_1 TestContainer_2 --group=daos_ci_test_grp_1@ --user=daos_ci_tester_1@'
2024-02-28 02:15:40,211 process          L0416 DEBUG| [stdout] Successfully set owner for container TestContainer_2
```

**Review by rpadma2 — APPROVED** [review]
LGTM.

**Review by phender — APPROVED** [review]

**Comment by daltonbohning** [comment]
Only ran the modified functional test and it passed: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13854/5/testReport/FTEST_deployment/

**Pull request merged**